engine/src/cmd/damageable.cpp

```diff
@@ -524,7 +524,7 @@ float Damageable::RShieldData() const {
 void Damageable::ArmorData(float armor[8]) const {
     Damageable *damageable = const_cast<Damageable *>(this);
     DamageableLayer armor_layer = damageable->GetArmorLayer();
-    for (int i = 0; i < 8; i++) {
```

Comment:
Yeah. You caught a bug of mine here. Did this segfault or fill the array with junk?

Reply:
I'm not sure if I actually hit this section of code during my testing. If I had, though, I think a segfault would have been most likely. The destination array was too small to write 8 elements to.